book-content/chapters/03-typescript-in-the-development-pipeline.md

```diff
@@ -0,0 +1,244 @@
+We've explored the relationship between JavaScript and TypeScript, and also how TypeScript improves your life as a developer. But let's go a bit deeper. In this chapter we'll get the TypeScript CLI up and running, and see how it fits into the development pipeline.
+
+As an example, we'll be looking at using TypeScript to build a web application. But TypeScript can also be used anywhere JavaScript can - in a Node, Electron, React Native or any other app.
+
+## The Problem with TypeScript in the Browser
+
+Consider this TypeScript file `example.ts` that contains a `run` function that logs a message to the console:
+
+```ts twoslash
+const run = (message: string) => {
+  console.log(message);
+};
+
+run("Hello!");
+```
+
+Alongside the `example.ts` file, we have a basic `index.html` file that references the `example.ts` file in a script tag.
+
+```html
+<!DOCTYPE html>
+
+<html lang="en">
+  <head>
+    <meta charset="UTF-8" />
+
+    <title>My App</title>
+  </head>
+
+  <body>
+    <script src="example.ts"></script>
+  </body>
+</html>
+```
+
+However, when we try to open the `index.html` file in a browser, you'll see an error in the console:
+
+```
+Unexpected token ':'
+```
+
+There aren't any red lines in the TypeScript file, so why is this happening?
+
+### Runtimes Can't Run TypeScript
```

Comment:
I guess, I'd expect mention of Deno (and maybe Bun) here to show that some can't run TypeScript. YMMV as maybe you don't want to go into too much details depending on the target audience